lib/executor.js

```diff
@@ -290,6 +290,9 @@ function setupMiddleware(app, instructions) {
 
 function setupComponents(app, instructions) {
   instructions.components.forEach(function(data) {
+    if (data === undefined)
+      return;
+
```

Comment:
Please filter out `undefined` records inside the compiler.

A possible implementation, sorry for giving you a wrong one initially:

```
return Object.keys(...)
  .filter(..check config is truthy..)
  .map(..original map function..)
```